src/main/java/jenkins/plugins/logstash/persistence/BuildData.java

```diff
@@ -65,7 +64,7 @@
  */
 public class BuildData {
   // ISO 8601 date format
-  public transient static final DateFormat DATE_FORMATTER = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
+  public transient static final FastDateFormat DATE_FORMATTER = FastDateFormat.getInstance("yyyy-MM-dd'T'HH:mm:ss.SSSZ");
```

Comment:
I'm not sure.. the addition of milliseconds can be a breaking change if somebody is parsing the date with a strict parser. I'd suggest extracting the format to configuration if you need the milis part

Reply:
ok, I will leave the format as is for now. Making this configurable could be done with the refactoring of configuration